app/views/govuk_component/docs/button.yml

```diff
@@ -0,0 +1,52 @@
+name: "Button"
+description: "Use buttons to move though a transaction, aim to use only one button per page."
+body: |
+  Button text should be short and describe the action the button performs.
+
+  GOV.UK Elements has more information on how buttons should be used https://govuk-elements.herokuapp.com/buttons/
+
+  Note: We do not consume GOV.UK Elements directly due to the naming conventions being leaky,
+  in time this component will be a wrapper for the [GOV.UK Frontend](https://github.com/alphagov/govuk-frontend) project's button component.
+
+  Global button styles seen at `app/assets/stylesheets/helpers/_buttons.scss`
+  should be considered deprecated with this component replacing any instances.
+
+  govuk-* namespace replaced with pub-c-* otherwise we will conflict with https://github.com/alphagov/govuk-frontend/blob/master/docs/components.md
+
+  This component is also extended for use in govspeak (http://govuk-component-guide.herokuapp.com/components/govspeak/fixtures/button).
+
+  These instances of buttons are added by Content Designers, ideally this duplication would not exist but we currently don't have shared markup
+  via our components within the generated [govspeak](https://github.com/alphagov/govspeak).
+  (This is a challenge to the reader :))
+accessibility_criteria: |
+  The button must:
+
+  - accept focus
+  - be focusable with a keyboard
+  - indicate when it has focus
+  - activate when focused and space is pressed
+  - activate when focused and enter is pressed
+  - have a role of button
```

Comment:
I took this as to say, the role is defined by default if you use a button element, whereas the role for an anchor needs to be added by specifying the role attribute.